infra/modules/gcp-psoxy-bulk/main.tf

```diff
@@ -69,30 +69,34 @@ resource "google_storage_bucket_iam_member" "grant_sa_read_on_processed_bucket"
   role   = "roles/storage.objectViewer"
 }
 
-# TODO: revisit if custom role is a good idea; this triggers security events for some orgs
-resource "google_project_iam_custom_role" "bucket-write" {
-  project     = var.project_id
-  role_id     = "writeAccess"
-  title       = "Access for writing and update objects in bucket"
-  description = "Write and update support, because storage.objectCreator role only support creation -not update"
-  permissions = ["storage.objects.create", "storage.objects.delete"]
-
-  lifecycle {
-    ignore_changes = [
-      labels
-    ]
-  }
-}
 
 resource "google_storage_bucket_iam_member" "access_for_processed_bucket" {
   bucket = google_storage_bucket.output-bucket.name
-  role   = google_project_iam_custom_role.bucket-write.id
+  role   = var.bucket_write_role_id
   member = "serviceAccount:${google_service_account.service-account.email}"
 }
 
+locals {
+  secret_bindings = merge({
+    PSOXY_SALT = {
+      secret_id      = var.salt_secret_id
+      version_number = var.salt_secret_version_number
+    }
+  }, var.secret_bindings)
+}
+
+resource "google_secret_manager_secret_iam_member" "grant_sa_accessor_on_secret" {
+  for_each = local.secret_bindings
+
+  project   = var.project_id
+  secret_id = each.value.secret_id
+  member    = "serviceAccount:${google_service_account.service-account.email}"
+  role      = "roles/secretmanager.secretAccessor"
+}
+
 resource "google_cloudfunctions_function" "function" {
```

Comment:
doubt: are we going to have the same issue with the tags? maybe we should add `lifecycle ignore_changes` everywhere too